File: Python3/647.Palindromic_Substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        total = 0

        for i in range(len(s)):
            total += self.extendOut(s, i, i)
            if (i < len(s) - 1):
                total += self.extendOut(s, i, i + 1)
        
        return total
        
    def extendOut(self, s: str, l: int, r: int):
        count = 0
        while (l >= 0 and r < len(s) and s[l] == s[r]):
            l -= 1
            r += 1
            count += 1
        return count
        

File: Java/647.Palindromic_Substrings.java
class Solution {
    public int countSubstrings(String s) {
        int total = 0;

        for (int i = 0; i < s.length(); ++i) {
            total += extendOut(s, i, i);
            if (i != s.length() - 1) {
                total += extendOut(s, i, i + 1);
            }
        }

        return total;
    }

    private int extendOut(String s, int l, int r) {
        int count = 0;
        while (l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)) {
            l--;
            r++;
            count++;
        }

        return count;
    }
}